pkg/controller/nodeipam/ipam/cidrset/cidr_set.go

```diff
@@ -30,15 +30,24 @@ import (
 // be allocated from.
 type CidrSet struct {
 	sync.Mutex
+	// clusterCIDR is the CIDR assigned to the cluster
 	clusterCIDR     *net.IPNet
+	// clusterIP is the network IP of the cluster
 	clusterIP       net.IP
```

Comment:
I had to go look at how this was used (since "cluster IP" usually means services).  It seems that this is initialized from `clusterCIDR.IP` - so I think we shoul djust get rid of this field entriely and just reference it as `clusterCIDR.IP`.  As a distinct commit or even PR, of course.

Reply:
let's continue on this, I just checked and removing clusterIP improves the performance